lib/AST/Decl.cpp

```diff
@@ -4206,10 +4206,8 @@ bool ValueDecl::hasOpenAccess(const DeclContext *useDC) const {
 
 bool ValueDecl::bypassResilienceInPackage(ModuleDecl *accessingModule) const {
   return getASTContext().LangOpts.EnableBypassResilienceInPackage &&
-          getModuleContext()->inSamePackage(accessingModule) &&
-          !getModuleContext()->isBuiltFromInterface() &&
-          getFormalAccessScope(/*useDC=*/nullptr,
-                               /*treatUsableFromInlineAsPublic=*/true).isPackage();
+         getModuleContext()->inSamePackage(accessingModule) &&
+         !getModuleContext()->isBuiltFromInterface();
```

Comment:
It doesn't need to be in this PR, but I'd like to propose that we make this check stronger to avoid miscompiles resulting from confusion over whether a binary module has been built with enough information to support non-resilient access. For modules built with `-experimental-skip-non-exportable-decls`, checking whether the binary module was built from interface is not sufficient because those binary modules behave as if they were built from interface.

I see a parallel between `-enable-testability` and this feature. I think that we should require that modules that want to be capable of providing non-resilient access to their `package` decls be built with an extra flag specified. We should then serialize that flag into the binary module and then check it on the consuming side before allowing non-resilient package access, and we should make sure that the behavior of `-experimental-skip-non-exportable-decls` is disabled when the package non-resilience flag is specified.

Reply:
I'll follow up on this in a separate PR as it needs changes in Module format.